table/block_based_table_reader.cc

```diff
@@ -2872,6 +2872,8 @@ void BlockBasedTable::MultiGet(const ReadOptions& read_options,
       iiter_unique_ptr.reset(iiter);
     }
 
+    std::unique_ptr<DataBlockIter> biter(new DataBlockIter());;
```

Comment:
Can we avoid this allocation? We used to put it in stack and now we regress to heap allocation.

Reply:
@siying Curious what's the problem with DataBlockIter being on heap here?

Reply:
@sagar0 we need to call malloc. It's something we general want to avoid in critical paths.

Reply:
Looks like if we have a `Reset()` method for `DataBlockIter`, then we do not have to perform dynamic memory allocation.

Reply:
Oh that's a good point. Let me see how to avoid it

Reply:
It appears that in `BlockBasedTableIterator`, we don't even call Reset(), and just call NewDataBlockIterator<DataBlockIter>() to move on to a new data block.